joint_trajectory_controller/src/joint_trajectory_controller_parameters.yaml

```diff
@@ -0,0 +1,73 @@
+joint_trajectory_controller:
+  joints: {
+    type: string_array,
+    default_value: [],
+    description: "Names of joints used by the controller",
+    # TODO(tylerjw): fix tests to remove initalization without joints
+    # validation: {
+    #   size_gt<>: [0],
+    # }
+  }
+  command_interfaces: {
+    type: string_array,
+    default_value: [],
+    description: "Names of command interfaces to claim",
+    validation: {
+      # TODO(tylerjw): fix tests to remove initalization without command_interfaces
+      # size_gt<>: [0],
+      unique<>: null,
+      subset_of<>: [["postion", "velocity", "acceleration", "effort",]],
+      command_interface_type_combinations: null,
+    }
+  }
+  state_interfaces: {
+    type: string_array,
+    default_value: [],
+    description: "Names of state interfaces to claim",
+    validation: {
+      # TODO(tylerjw): fix tests to remove initalization without state_interfaces
+      # size_gt<>: [0],
+      unique<>: null,
+      subset_of<>: [["postion", "velocity", "acceleration",]],
+      state_interface_type_combinations: null,
+    }
+  }
+  allow_partial_joints_goal: {
+    type: bool,
+    default_value: false,
+    description: "Goals with partial set of joints are allowed",
+  }
+  open_loop_control: {
+    type: bool,
+    default_value: false,
+    description: "Run the controller in open-loop, i.e., read hardware states only when starting controller. This is useful when robot is not exactly following the commanded trajectory.",
+  }
+  allow_integration_in_goal_trajectories: {
+    type: bool,
+    default_value: false,
+    description: "Allow integration in goal trajectories to accept goals without position or velocity specified",
+  }
+  state_publish_rate: {
+    type: double,
+    default_value: 50.0,
+    description: "Rate controller state is published",
+    validation: {
+      lower_bounds: [0.1]
+    }
+  }
+  action_monitor_rate: {
+    type: double,
+    default_value: 20.0,
+    description: "Rate status changes will be monitored",
+    validation: {
+      lower_bounds: [0.1]
+    }
+  }
+  interpolation_method: {
+    type: string,
+    default_value: "splines",
+    description: "The type of interopoation to use if at all",
+    validation: {
+      one_of<>: [["splines", "none"]],
```

Comment:
This is nice

Reply:
With this validation, if you accidentially set this value to something else you'll get a helpful error message that says something like this:

```Parameter failed validation: 'bla' set for 'interpolation_method' is not in the set ['splines', 'none']```

Reply:
OK. Well, that case is already covered and it even allows for a default.

```
[[nodiscard]] inline InterpolationMethod from_string(const std::string & interpolation_method)
{
  if (interpolation_method.compare(InterpolationMethodMap.at(InterpolationMethod::NONE)) == 0)
  {
    return InterpolationMethod::NONE;
  }
  else if (
    !interpolation_method.compare(
      InterpolationMethodMap.at(InterpolationMethod::VARIABLE_DEGREE_SPLINE)) == 0)
  {
    return InterpolationMethod::VARIABLE_DEGREE_SPLINE;
  }
  // Default
  else
  {
    RCLCPP_INFO_STREAM(
      LOGGER,
      "No interpolation method parameter was given. Using the default, VARIABLE_DEGREE_SPLINE.");
    return InterpolationMethod::VARIABLE_DEGREE_SPLINE;
  }
}
```

Reply:
I think I should be able to remove that code from the `from_string` as with this you won't ever be able to get to the code path where it is invalid.